packages/shared/lib/services/config.service.ts

```diff
@@ -167,7 +168,7 @@ class ConfigService {
         return { id: id[0]?.id, unique_key: config.unique_key } as Pick<ProviderConfig, 'id' | 'unique_key'>;
     }
 
-    async deleteProviderConfig(providerConfigKey: string, environment_id: number): Promise<number> {
+    async deleteProviderConfig(providerConfigKey: string, environment_id: number, orchestrator: Orchestrator): Promise<number> {
```

Comment:
here we have a good example of a dependency being called deep down into the codebase and requires the dependency to be passed along a chain of function calls. This is not pretty and not really sustainable on the long term but I am not gonna refactor now all the services that depends on temporal/orchestrator

Reply:
We know it's mostly because of the CLI (and because we forced ourselves to not publish any packages until we fix things). Any refactor will be stuck by the fact that CLI still depends on shared

Reply:
It is not just that. There is a few basic programming principles that we don't follow much in the codebase. For example too many functions are doing more than one thing. This one is a good example, it updates the db, call deleteSyncFiles, also call deleteSyncs.
Couple that with too many globally accessible components and you ends up with highly couple subsystems that are hard to modify/refactor :(